src/pipelines/step/composite/mod.rs

```diff
@@ -0,0 +1,148 @@
+//! This module defines composite steps that can be used to compose multiple
+//! steps into a single step.
```

Comment:
I'd probably keep this work scoped to `src/steps/utils.rs` or somewhere there. I'm not sure if we will need to modify the core library to enable combinator/composite steps.